elasticsearch/pom.xml

```diff
@@ -70,25 +47,33 @@
     </resources>
     <plugins>
       <plugin>
-        <artifactId>maven-compiler-plugin</artifactId>
-        <configuration>
-          <fork>true</fork>
-          <compilerArgs>
-            <arg>-J-Dproject.version=${project.version}</arg>
-          </compilerArgs>
-        </configuration>
-      </plugin>
-      <plugin>
         <groupId>io.fabric8</groupId>
-        <artifactId>docker-maven-plugin</artifactId>
-        <version>${docker.maven.plugin.version}</version>
+        <artifactId>fabric8-maven-plugin</artifactId>
+        <version>${fabric8.maven.plugin.version}</version>
+        <executions>
+          <execution>
+            <id>resource</id>
+            <goals>
+              <goal>resource</goal>
+              <goal>build</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>json</id>
+            <phase>none</phase>
```

Comment:
Maybe sth like

```xml
<profiles>
  <profile>
    <id>f-m-p-3</id>
    <activation>
      <property>
        <name>fabric8-maven-plugin.majorVersion</name>
        <value>3</value>
      </property>
    </activation>
    ...
  </profile>
</profiles>
```

in the parent pom, with two profiles for each variant. An then set in each individual project pom.xml `fabric8-maven-plugin.majorVersion` to `2` or `3`.